base/expr.jl

```diff
@@ -184,6 +184,57 @@ macro noinline(ex)
     esc(isa(ex, Expr) ? pushmeta!(ex, :noinline) : ex)
 end
 
+"""
+    @pure
+
+Annotates a function to ease type inference by strictly specifying the
+output is completely determined by the input.
+
+Note: Misuse can lead to regressions.
+
+Do not use if the function involved:
+
+- Involves globals, pointers
+- Recurses
+- Does not return exactly (`===`) the same result for the same input
+- Gets its methods extended after it is called
+
```

Comment:
- Uses dispatch on one of its arguments.

Reply:
Thanks!
Clarifying: one, or any of its arguments?

Reply:
Yes, any.